From 24909a09fb9650e78f909ee1269d276645260073 Mon Sep 17 00:00:00 2001
From: bilux <i.bilux@gmail.com>
Date: Mon, 23 Mar 2020 09:37:30 +0100
Subject: [PATCH] egl: Add back b/10194508 workaround

This was removed because it did not pass CTS and was applied across the board for all OMAP4 devices, but some still-shipping OMAP4 devices don't need it due to using DDK 1.12 drivers. Those drivers are only for OMAP4470 though, and other devices besides OMAPs depend on this workaround too, and we don't care about CTS for them anyway.

This patch adds this previously-long-standing behavior (EGL_NATIVE_VISUAL_ID) back under the opt-in BOARD_EGL_WORKAROUND_BUG_10194508 flag.

Signed-off-by: bilux <i.bilux@gmail.com>
---
 opengl/libs/EGL/eglApi.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/opengl/libs/EGL/eglApi.cpp b/opengl/libs/EGL/eglApi.cpp
index 3312b03..3a233e8 100644
--- a/opengl/libs/EGL/eglApi.cpp
+++ b/opengl/libs/EGL/eglApi.cpp
@@ -15,6 +15,7 @@
  */
 
 #define ATRACE_TAG ATRACE_TAG_GRAPHICS
+#define WORKAROUND_BUG_10194508
 
 #include <ctype.h>
 #include <dlfcn.h>
@@ -722,7 +723,17 @@ EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
 
         EGLDisplay iDpy = dp->disp.dpy;
         android_pixel_format format;
+
+#ifdef WORKAROUND_BUG_10194508
+        if (!cnx->egl.eglGetConfigAttrib(iDpy, config, EGL_NATIVE_VISUAL_ID,
+                &format)) {
+            ALOGE("eglGetConfigAttrib(EGL_NATIVE_VISUAL_ID) failed: %#x",
+                    eglGetError());
+            format = 0;
+        }
+#else
         getNativePixelFormat(iDpy, cnx, config, &format);
+#endif
 
         // now select correct colorspace and dataspace based on user's attribute list
         EGLint colorSpace = EGL_UNKNOWN;
-- 
2.25.1

